core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -209,6 +211,11 @@ class BrokerServer(
 
       val controllerNodes = RaftConfig.voterConnectionsToNodes(controllerQuorumVotersFuture.get()).asScala
       val controllerNodeProvider = RaftControllerNodeProvider(raftManager, config, controllerNodes)
+      val currentNodeControllerApiVersions = if (config.isKRaftCoResidentMode) {
+        Some(new NodeApiVersions(ApiKeys.controllerApis().asScala.map(ApiVersionsResponse.toApiVersion).asJava))
+      } else {
+        None
```

Comment:
Unless I'm misreading the code, it seems like this was not None before. Are there implications for adding the None case?

Reply:
The `NodeApiVersions` is used in `BrokerToControllerChannelManager` to get controller ApiVersion only if the current node is both the broker and the controller(which only happen in KRaft co-resident mode or zk mode). in other cases, it is not used so I just set it to None, and it will make no difference if I change it to Some(controllerApiVersion)